sky/data/sky_csync.py

```diff
@@ -0,0 +1,337 @@
+        else:
+            # successfully completed sync process
+            break
+    else:
+        raise RuntimeError(f'Failed to sync {src_to_bucket} after '
+                           f'{max_retries} number of retries. Check '
+                           'the log file in ~/.sky/ for more'
+                           'details') from None
+
+
+@main.command()
+@click.argument('source', required=True, type=str)
+@click.argument('storetype', required=True, type=str)
+@click.argument('destination', required=True, type=str)
+@click.option('--num-threads', required=False, default=10, type=int, help='')
+@click.option('--interval-seconds',
+              required=False,
+              default=600,
+              type=int,
+              help='')
+@click.option('--delete',
+              required=False,
+              default=False,
+              type=bool,
+              is_flag=True,
+              help='')
+@click.option('--no-follow-symlinks',
+              required=False,
+              default=False,
+              type=bool,
+              is_flag=True,
+              help='')
+def csync(source: str, storetype: str, destination: str, num_threads: int,
+          interval_seconds: int, delete: bool, no_follow_symlinks: bool):
+    """Runs daemon to sync the source to the bucket every INTERVAL seconds.
+
+    Creates an entry of pid of the sync process in local database while sync
+    command is runninng and removes it when completed.
+
+    Args:
+        source (str): The local path to the directory that you want to sync.
+        storetype (str): The type of cloud storage to sync to.
+        destination (str): The bucket or subdirectory in the bucket where the
+            files should be synced.
+        num_threads (int): The number of threads to use for the sync operation.
+        interval_seconds (int): The time interval, in seconds, at which to run
+            the sync operation.
+        delete (bool): Whether or not to delete files in the destination that
+            are not present in the source.
+        no_follow_symlinks (bool): Whether or not to follow symbolic links in
+            the source directory.
+    """
+    full_src = os.path.abspath(os.path.expanduser(source))
+    # If the given source is already mounted with CSYNC, terminate it.
+    if _get_csync_pid_from_source_path(full_src):
+        _terminate([full_src])
+    csync_pid = os.getpid()
+    _add_running_csync(csync_pid, full_src)
+    while True:
+        start_time = time.time()
+        run_sync(full_src, storetype, destination, num_threads,
+                 interval_seconds, delete, no_follow_symlinks, csync_pid)
+        end_time = time.time()
+        # Given the interval_seconds and the time elapsed during the sync
+        # operation, we compute remaining time to wait before the next
+        # sync operation.
+        elapsed_time = int(end_time - start_time)
+        remaining_interval = max(0, interval_seconds - elapsed_time)
```

Comment:
Shall we sleep for some minimal intervals (like 3s) to avoid continuously calling API?